### PR TITLE
Add compressedByteProvider

### PR DESCRIPTION
**Pull request by JonathanMeans**

### CODE REVIEW

**Comment by asylvest** [comment]
I don't see a unit test for the compressed byte provider... Can you add one?  Also, a simple program which writes out a compressed NITF using the byte provider would show people how to use it - could update one of the existing programs which writes out a NITF from scratch to use it.  Can hand in just one big buffer for that one.

**Comment by JonathanMeans** [comment]
Done

File: modules/c++/nitf/include/nitf/ByteProvider.hpp [hunk 1]
```diff
@@ -207,9 +208,8 @@ class ByteProvider
      * blocked memory layout, including pad pixels for partial blocks.
      *
      * \param imageData The image data pixels to write.  The underlying type
-     * will be complex short or complex float based on the complex data sent
-     * in during initialize.  Must be in big endian order and blocked if the
-     * NITF is.
+     * will be a single byte per pixel. Must be in big endian order and
```
Comment (asylvest) [inline modules/c++/nitf/include/nitf/ByteProvider.hpp]: This is the base class - it doesn't know how many bytes/pixel it is...

File: modules/c++/nitf/include/nitf/CompressedByteProvider.hpp [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+     * means that on a little endian system, you must byte swap the pixel data
+     * prior to calling this method.
+     *
+     * \note This method does not perform any blocking on the pixel data.  If
+     * the NITF is blocked, the pixel data must already be in a contiguous
+     * blocked memory layout, including pad pixels for partial blocks.
+     *
+     * \param imageData The image data pixels to write.  The underlying type
+     * will be complex short or complex float based on the complex data sent
+     * in during initialize.  Must be in big endian order and blocked if the
```
Comment (asylvest) [inline modules/c++/nitf/include/nitf/CompressedByteProvider.hpp]: This comment still needs to get updated

File: modules/c++/nitf/tests/test_create_nitf_with_byte_provider.cpp [hunk 3]
```diff
@@ -0,0 +1,1214 @@
...
+  "\235\235\246\232\232\243\221\221\232\203\203\213yyzzzz\225\225\225\303\303"
+  "\303\351\351\351\366\366\366\366\366\366\357\357\357\347\347\347\335\335"
+  "\335\324\324\324\315\315\315\306\306\306\303\303\303\303\303\303\300\300"
+  "\302\245\245\257\243\243\254pp\206oo\205oo\205oo\205oo\205oo\205oo\205\236"
+  "\236\247\234\234\245\256\256\260\251\251\251\235\235\235\223\223\223\215"
+  "\215\215\211\211\211\210\210\210\211\211\211\216\216\216\232\232\232\252"
+  "\252\252\300\300\300\322\322\322\344\344\344\366\366\366\376\376\376\377"
+  "\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377\377"
+  "\377\377\377\377\377\377\377\377\377\377\377\377\377\377\376\376\376\371"
+  "\371\37
```
Comment (asylvest) [inline modules/c++/nitf/tests/test_create_nitf_with_byte_provider.cpp]: Shouldn't this be `"M"` or something else?

**Pull request merged**